cabal-install/Distribution/Client/InstallPlan.hs

```diff
@@ -223,17 +223,21 @@ type InstallPlan = GenericInstallPlan
 
 -- | Smart constructor that deals with caching the 'Graph' representation.
 --
-mkInstallPlan :: Graph (GenericPlanPackage ipkg srcpkg)
+mkInstallPlan :: (IsUnit ipkg, IsUnit srcpkg)
```

Comment:
I thought of that but couldn't be bothered with the CPP required given the call stack stuff only works with newer ghcs.